src/main/java/ir/sahab/kafkaconsumer/SmartCommitKafkaConsumer.java

```diff
@@ -132,13 +131,19 @@
      * We have used a same callback that does nothing other than writing simple logs and resetting
      * the {@link #offsetTracker}.
      */
-    private final ConsumerRebalanceListener rebalanceListener;
+    private final ConsumerRebalanceListener internalRebalanceListener;
 
     /**
      * List of partitions assigned to consumer after first connection to Kafka or rebalancing. Assigned partitions
      * cached and will be used later.
      */
-    private List<TopicPartition> assignedPartitions;
+    private final List<TopicPartition> assignedPartitions;
+
+    /**
+     * The callback which is called when rebalance happens.
```

Comment:
Replace:
The callback ...  ->  The callback provided by the client ...

Reply:
Done